### PR TITLE
Move `create account` to direct function call

### PR DESCRIPTION
**Pull request by janezpodhostnik**
closes: dapperlabs/flow-internal#1513

CreateAccount was changed from a meta transaction to a contract function call.

RestrictedAccountCreationEnabled condition was moved from the fvm to the FlowServiceAccount contract https://github.com/onflow/flow-core-contracts/pull/187. Because of this setting this parameter for tests had to be moved to the bootstrapping procedure.

:exclamation:  TODO before merge: change `flow-core-contracts` to point to a tagged version.

### CODE REVIEW

File: module/mock/job_consumer.go [hunk 1]
```diff
@@ -23,14 +23,14 @@ func (_m *JobConsumer) NotifyJobIsDone(_a0 module.JobID) {
 }
 
 // Size provides a mock function with given fields:
-func (_m *JobConsumer) Size() int {
```
Comment (janezpodhostnik) [inline module/mock/job_consumer.go:26]: Not sure why this file is committed. This change was made by re-generating the mocks. Should I revert the change or remove this file entirely?

**Review by turbolent — APPROVED** [review]
Nice work!

File: module/mock/job_consumer.go (hunk 1 above)
Comment (ramtinms) [reply module/mock/job_consumer.go:26]: it should be fine, it happens from time to time that a mock was not commited, I think we should keep this change.

File: fvm/env.go [hunk 2]
```diff
@@ -1050,21 +1051,25 @@ func (e *transactionEnv) CreateAccount(payer runtime.Address) (address runtime.A
...
+			"setupNewAccount",
+			[]interpreter.Value{
+				interpreter.NewAddressValue(common.BytesToAddress(flowAddress.Bytes())),
+				interpreter.NewAddressValue(common.BytesToAddress(payer.Bytes())),
+			},
+			[]sema.Type{
+				sema.AuthAccountType,
+				sema.AuthAccountType,
+			},
+			e.ctx.Logger,
```
Comment (ramtinms) [inline fvm/env.go:1066]: I'm a bit worried about this, the rest of the context of how this method is set is not visible here, and might open up some space for future mistakes, wonder if we should put more comments or reference to the method definition in the core contracts.

**Review by ramtinms — APPROVED** [review]

File: fvm/env.go (hunk 2 above)
Comment (janezpodhostnik) [reply fvm/env.go:1066]: I will add more comments!
